ui/components/app/terms-of-use-popup/terms-of-use-popup.js

```diff
@@ -0,0 +1,1181 @@
+import PropTypes from 'prop-types';
+import { I18nContext } from '../../../contexts/i18n';
+import Popover from '../../ui/popover';
+import {
+  AlignItems,
+  FLEX_DIRECTION,
+  TextVariant,
+  Color,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import {
+  Text,
+  Button,
+  BUTTON_TYPES,
+  ButtonLink,
+  Label,
+} from '../../component-library';
+import Box from '../../ui/box';
+import CheckBox from '../../ui/check-box/check-box.component';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+
+export default function TermsOfUsePopup({ onAccept }) {
+  const t = useContext(I18nContext);
+  const trackEvent = useContext(MetaMetricsContext);
+  const [isTermsOfUseChecked, setIsTermsOfUseChecked] = useState(false);
+
+  const popoverRef = useRef();
+  const bottomRef = React.createRef();
+
+  const handleScrollDownClick = (e) => {
+    e.stopPropagation();
+    bottomRef.current.scrollIntoView({
+      behavior: 'smooth',
+    });
+  };
+
+  useEffect(() => {
+    trackEvent({
+      category: MetaMetricsEventCategory.Onboarding,
+      event: MetaMetricsEventName.TermsOfUseShown,
+      properties: {
+        location: 'Terms Of Use Popover',
+      },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <Popover
+      className="terms-of-use__popover"
+      popoverRef={popoverRef}
+      title={t('termsOfUseTitle')}
+      showScrollDown
+      onScrollDownButtonClick={handleScrollDownClick}
+      footerProps={{
+        justifyContent: AlignItems.center,
+        flexDirection: FLEX_DIRECTION.COLUMN,
+      }}
+      footer={
+        <>
+          <Button
+            type={BUTTON_TYPES.PRIMARY}
+            className="terms-of-use__button"
+            onClick={onAccept}
+            disabled={!isTermsOfUseChecked}
+            data-testid="terms-of-use-accept-button"
+          >
+            {t('accept')}
+          </Button>
+          <Text
+            as="p"
+            marginTop={4}
+            className="terms-of-use__footer-text"
+            color={TextColor.textAlternative}
+          >
+            {t('termsOfUseFooterText')}
+          </Text>
+        </>
+      }
+    >
+      <Box className="terms-of-use">
+        <Box
+          className="terms-of-use__content"
+          marginBottom={4}
+          marginLeft={4}
+          marginRight={4}
+        >
+          <Text variant={TextVariant.bodySm} marginBottom={4}>
+            IMPORTANT NOTICE: THIS AGREEMENT IS SUBJECT TO BINDING ARBITRATION
+            AND A WAIVER OF CLASS ACTION RIGHTS AS DETAILED IN SECTION 11.
+            PLEASE READ THE AGREEMENT CAREFULLY.
+          </Text>
+          <Text variant={TextVariant.bodySm} marginBottom={4}>
+            ConsenSys Software Inc. (“ConsenSys,” “we,” “us,” or “our”) is the
```

Comment:
Question: should we move them to language file as well?

Reply:
I wanted to highlight this point as well. But seems like we only want to do it in english https://github.com/MetaMask/metamask-extension/pull/18221#discussion_r1140373814

Reply:
Thanks for the link, it is clear :)

Reply:
We will have to revisit this in the future, whether that's adding the text to a constant file or using an iframe, as @kevinghim suggested. For now we are rolling with this.